### PR TITLE
Add Recipe documentation and update Task dev doc

### PR DESCRIPTION
**Pull request by aarontp**

### CODE REVIEW

File: docs/developer/developing-new-tasks.md [hunk 1]
```diff
@@ -66,48 +69,98 @@ little more explaining is this one:
...
+completed, the post-processor will tear down this state (e.g. unmount or
+detach, etc).  For more details on the different states and how the
+pre/post-processors will set these up at runtime, see the
+[`Evidence.preprocess()`
+docstrings](https://github.com/google/turbinia/blob/cc79288ae36cfec749381b80694b4c1290d76583/turbinia/evidence.py#L291).
+
+### Evidence Paths
+
+As mentioned above, the pre-processor that runs before the Task is executed
+will set the path `evidence.local_path` to point to the local evidence. If the
```
Comment (alimez) [inline docs/developer/developing-new-tasks.md:130]: s/evidence/Evidence/g

File: docs/developer/developing-new-tasks.md [hunk 2]
```diff
@@ -66,48 +69,98 @@ little more explaining is this one:
...
+was mounted on if it was mounted, etc).  See the [docstrings for these
+attributes in the Evidence
+object](https://github.com/google/turbinia/blob/cc79288ae36cfec749381b80694b4c1290d76583/turbinia/evidence.py#L127)
+for more details on the differences, but in summary, Tasks should use
+`.local_path` to process the incomming Evidence and `.source_path` for newly
+created Evidence.
+
+### Recipe configuration
+Tasks can also specify a set of variables that can be configured and set
+through [recipes](../user/recipes.md) recipes.  This allows users to pre-define set
```
Comment (alimez) [inline docs/developer/developing-new-tasks.md:144]: Remove the second recipes.

File: docs/developer/developing-new-tasks.md [hunk 3]
```diff
@@ -66,48 +69,98 @@ little more explaining is this one:
...
+configurations for the runtime behavior of Tasks along with which Jobs/Tasks
+should be run.  Each task has a `TASK_CONFIG` dictionary set at the object
```
Comment (alimez) [inline docs/developer/developing-new-tasks.md:146]: s/task/Task/g

File: docs/developer/developing-new-tasks.md [hunk 4]
```diff
@@ -66,48 +69,98 @@ little more explaining is this one:
...
+`TASK_CONFIG` as an example. Tasks can access these variables by referencing
+the dictionary at `self.task_config`.
 
 ## Boilerplate and Glue
 
 The only two interesting bits for the Job definition in
-`turbinia/jobs/strings.py` are this one that sets the input and output evidence
-types for the Task (so the Task Manager knows what kinds of Tasks to schedule):
+`turbinia/jobs/strings.py` are this one that sets the allowable input and
+output evidence types for the Task (so the Task Manager knows what kinds of
```
Comment (alimez) [inline docs/developer/developing-new-tasks.md:158]: s/evidence/Evidence/g

File: docs/user/recipes.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+turbiniactl --recipe_path ./recipes/triage.yaml googleclouddisk -d diskname-to-process
+```
+
+#### Writing new Recipes
+Recipes are `.yaml` files that are read and validated by the client and passed
+to the server along with the processing request data.  There are no required
+sections and they can contain a `globals` section and zero or more Task
+sections.  Each Task section must contain a `task:` key that references the
+relevant Task that the section applies to.  Other keys in either the `globals`
+or task sections must match the pre-defined keys for those sections.  Here is a
```
Comment (alimez) [inline docs/user/recipes.md:34]: s/task/`task`/g

File: docs/user/recipes.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  task: 'PlasoTask'
+  status_view: 'none'
+  hashers: 'all'
+  partition: 'all'
+  vss_stores: 'all'
+```
+
+For adding additional configuration options to a given Task, please see the
+[recipes configuration
+section](../developer/developing-new-tasks.md#recipe-configuration) in the
```
Comment (alimez) [inline docs/user/recipes.md:97]: Not sure why this fails, maybe give this a shot https://stackoverflow.com/questions/38987963/create-a-link-to-the-header-of-another-page-in-github

**Comment by aarontp** [comment]
For some reason it doesn't like the link directly to the header so I just removed it.

@alimez PTAL

**Review by alimez — APPROVED** [review]
LGTM

**Pull request merged**